### PR TITLE
Ard chapter14

### PR DESCRIPTION
**Pull request by ardavey**

### CODE REVIEW

File: ard/My-List-Util/t/shuffle.t [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+ok( defined &My::List::Util::shuffle, 'shuffle() is defined' );
+
+# Happy shuffling
+{
+  my @array = qw( a b c d e f g );
+  my @shuffled = My::List::Util::shuffle( @array );
+  is( scalar @array, scalar @shuffled, 'Shuffled list is the same size' );
+  isnt( "@array", "@shuffled", 'Shuffled array is different' );
```
Comment (utopia-2) [inline ard/My-List-Util/t/shuffle.t:17]: This still has a chance of producing the same ordering.

File: ard/My-List-Util/lib/My/List/Util.pm [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+=head2 sum
+
+Returns the sum of the numerics in the provided list
+
+=cut
+
+sub sum {
+  my $sum;
+  foreach my $num ( grep( /^-?\d*\.?\d*$/, @_ ) ) {
```
Comment (utopia-2) [inline ard/My-List-Util/lib/My/List/Util.pm:44]: Perl stringifies huge/tiny numbers using scientific notation, which this will throw away.

Comment (ardavey) [inline ard/My-List-Util/lib/My/List/Util.pm:44]: Doesn't it only do that upon printing?

Comment (utopia-2) [inline ard/My-List-Util/lib/My/List/Util.pm:44]: I'd thought the same, but my huge numbers test fails unless I include scientific notation in the filtering.

Comment (utopia-2) [inline ard/My-List-Util/lib/My/List/Util.pm:44]: The pure-perl version of `Scalar::Util::looks_like_number` does check for scientific notation, and a couple of other cases we all missed:

``` perl
  return 1 if (/^[+-]?[0-9]+$/); # is a +/- integer
  return 1 if (/^([+-]?)(?=[0-9]|\.[0-9])[0-9]*(\.[0-9]*)?([Ee]([+-]?[0-9]+))?$/); # a C float
  return 1 if ($] >= 5.008 and /^(Inf(inity)?|NaN)$/i) or ($] >= 5.006001 and /^Inf$/i);
```

Comment (ardavey) [inline ard/My-List-Util/lib/My/List/Util.pm:44]: This test case passes.

``` perl
is( My::List::Util::sum( qw( 1234567890123456789012345678901234567890 1234567890123456789012345678901234567890 1234567890123456789012345678901234567890 ) ), 3703703670370370367037037036703703703670, 'Huuuuuuuuuuge' );
```

Comment (thruddle) [inline ard/My-List-Util/lib/My/List/Util.pm:44]: This test case doesn't.

``` perl
is( My::List::Util::sum( 10**100, 2*10**100 ), 3*10**100, 'You call that huge?' );
```

**Pull request merged**